### ISSUE TITLE
Running "git log" or "git status" in Ubuntu WSL causes Windows Terminal to break buffer when scrolling up/down

### ISSUE THREAD

**Issue by tommai78101** [microsoft/terminal#8371]
# Environment

```none
Windows build number:  Win32NT 10.0.19042.0 (Microsoft Windows NT 10.0.19042.0)
Windows Terminal version (if applicable): Version: 1.4.3243.0

Any other software?
git version 2.17.1
```

# Steps to reproduce

1. Launch Windows Terminal.
2. Launch an Ubuntu WSL session. (Not WSL2, just WSL)
3. In Ubuntu WSL, find a directory location where there's a known cloned `.git` repository.
4. In the repository, run `git log` or `git status`, if you have a lot of modified changes yet to be committed. This should at least print enough characters in Windows Terminal, that you will definitely need to scroll up/down to view everything that was printed in the window.
5. Use your mouse and drag the Windows Terminal scrollbar up or down very quickly.
6. Observe the error. I think the display is not refreshing correctly. (See below)

# Expected behavior

I should be able to view text normally.

# Actual behavior

![GIF](https://user-images.githubusercontent.com/6444116/99931811-03c13b80-2d24-11eb-9bc1-d2b545d98076.gif)

**Comment by tommai78101** [comment]
I made it easier to reproduce, instead of seeing all those red text which is really hard to view:

![GIF](https://user-images.githubusercontent.com/6444116/99932024-bb564d80-2d24-11eb-8daa-637aeedaa8d5.gif)

**Comment by DHowett** [comment]
Whoa! What GPU do you have? We only used to see this when partial presentation was broken...

**Comment by skyline75489** [comment]
And does `experimental.rendering.software` or `experimental.rendering.forceFullRepaint` make a difference here? 

See: https://docs.microsoft.com/en-us/windows/terminal/customize-settings/global-settings#rendering-settings

**Comment by zadjii-msft** [comment]
Or, does selecting (and deselecting) the text in the buffer reveal what you'd actually expect to be visible?

**Comment by tommai78101** [comment]
> Whoa! What GPU do you have? We only used to see this when partial presentation was broken...

Dual GPUs setup. One GTX 1060, and one RTX 2070 Turbo.

> And does `experimental.rendering.software` or `experimental.rendering.forceFullRepaint` make a difference here?
> 
> See: https://docs.microsoft.com/en-us/windows/terminal/customize-settings/global-settings#rendering-settings

Just the setting `"experimental.rendering.forceFullRepaint": true` is good enough for me to fix the issue. I didn't know about this setting configuration before.

> Or, does selecting (and deselecting) the text in the buffer reveal what you'd actually expect to be visible?

Selecting and deselecting the text in the buffer reveals the text I expected it to print out after doing a simple `ls -a`.

**Comment by zadjii-msft** [comment]
Out of curiosity, does this bug only repro for one of the GPUs? If `forceFullRepaint` fixes this, then it might be a sign that either partial presentation doesn't work for that GPU, or we're not doing something right for that specific one.

**Comment by tommai78101** [comment]
> `Repaint` fixes this, then it might be a sign that either partial presentation doesn't work for that GPU, or we're not doing something right for that specific one.

How should I test this? Do I just disable the device, then reboot for each of these graphics card?

![image](https://user-images.githubusercontent.com/6444116/100014953-2a738680-2da5-11eb-80f2-85e1ceaa3564.png)

**Comment by tommai78101** [comment]
Actually, another question: How do you determine what rendering device the Windows Terminal is using for rendering? Maybe if that information is included somewhere when creating a diagnostic file, would it help to pinpoint which device the rendering is malfunctioning?

If these are not viable, then nevermind.

**Comment by onomatopellan** [comment]
@tommai78101 I don't have more than GPU but did you try [this](https://www.howtogeek.com/351522/how-to-choose-which-gpu-a-game-uses-on-windows-10)?

**Comment by tommai78101** [comment]
@onomatopellan That's actually better. Let me try that and see how it goes.

**Comment by tommai78101** [comment]
@onomatopellan @zadjii-msft 

Apparently, Windows Terminal runs on RTX 2070, and that's what's causing the screen tearing/partial rendering issue when scrolling up/down fast enough.

![image](https://user-images.githubusercontent.com/6444116/100018221-226a1580-2daa-11eb-9072-08b132e3441b.png)

Also it is strange that my GTX 1060 is gone from this menu...

**Comment by tommai78101** [comment]
And if I switch to Intel UHD Graphics 630, I don't get that screen tearing.

![GIF](https://user-images.githubusercontent.com/6444116/100018451-7d037180-2daa-11eb-9b60-edb938d54703.gif)

**Comment by onomatopellan** [comment]
@tommai78101 It seems you need to specify which one is the High Performance GPU. Can you find somethinng like this?
 
![img](https://46c4ts1tskv22sdav81j9c69-wpengine.netdna-ssl.com/wp-content/uploads/prod/sites/2/2020/08/16635566f758c021e6b341dfba58769d.jpg)

**Comment by tommai78101** [comment]
> @tommai78101 It seems you need to specify which one is the High Performance GPU. Can you find something like [this]

No, I'm not on the Windows 10 Insider, since this is is home personal computer. I do not want to be an Insider.

I'm still on the 19042 build, per OP at the top.

**Comment by DHowett** [comment]
I've marked this bug up for our backlog and stuck the "needs repro" tag on it. Probably our best bet is gonna be someone on the team getting this under a debugger . . . but that's going to be difficult.

I think there might be some tracing we can use to help determine what exactly is happening, but I'll need to consult.

**Comment by tommai78101** [comment]
Cool, let me know if you need anything.

For now, I'm going to stick to using the `experimental.rendering.forceFullRepaint` option, since it works perfectly.

**Comment by zadjii-msft** [comment]
Hey so I know it's been a few years - You still seeing this/? If you are, could you try out the [new rendering engine](https://docs.microsoft.com/en-us/windows/terminal/customize-settings/profile-advanced#experimental-text-rendering-engine) in the [Preview builds](https://apps.microsoft.com/store/detail/windows-terminal-preview/9N8G5RFZ9XK3?hl=en-us&gl=US)? I'm curious if that new engine just fixes this for you.

**Comment by tommai78101** [comment]
@zadjii-msft Hello, I installed the Windows Terminal Preview, and just checked. Unfortunately, the issue is still not fixed when I turned off the `experimental.rendering.forceFullRepaint` option.

**Comment by tommai78101** [comment]
In action:


![GIF](https://user-images.githubusercontent.com/6444116/184395559-9ac64346-2634-4ee5-b805-8c7f3ffb7fe1.gif)

**Comment by zadjii-msft** [comment]
Ah okay that's my bad. The `experimental.useAtlasEngine` setting actually goes in your `profile` settings, rather than the globals (like the other rendering settings). Can you try there instead? (also, that setting doesn't hot-reload, so you'll need to reboot the terminal).

Thanks for bearing with me!

**Comment by tommai78101** [comment]
@zadjii-msft I moved the settings under the `profile` setting, and redid the test.  The new rendering engine settings seems to be working now.

Seems to be working now. 

![GIF](https://user-images.githubusercontent.com/6444116/184412468-bf6be065-50cd-4c73-9dde-82ec1abd9712.gif)

**Comment by zadjii-msft** [comment]
> What was the issue found that would cause the error with the rendering issue before the new Atlas engine?

Honestly, I don't know! The two engines have pretty different paths by which they interact with the GPU. Perhaps the set of calls we're using with the old renderer (most likely `Present1`) just don't play nice with your GPU. 

This bumps a discussion topic we've been putting off. What exactly should we do with bugs like this? Where there's a bug in the dxrenderer, but it's resolved in the new renderer. We're ultimately planning on defaulting to the new engine for everyone (once it's a bit more stable). 

I'm proposing we add a `Resolution-FixedByAtlasRenderer` label. Use that to tag issues we know are fixed already. Maybe just close those out right now? Maybe we could add a rule to the bot so that `/atlas` would give a little blurb about the new renderer and how to enable it for now.

**Comment by zadjii-msft** [comment]
Oh we looped back on this but never actually closed it out. 🤦 

We did end up tracking a whole pile of "fixed by atlas" things, but this escaped.
